makeflow/src/makeflow.c

```diff
@@ -950,9 +996,10 @@ static void makeflow_run( struct dag *d )
         }
 
 	while(!makeflow_abort_flag) {
+		did_find_archived_job = 0;
 		makeflow_dispatch_ready_jobs(d);
 
-		if(dag_local_jobs_running(d)==0 && dag_remote_jobs_running(d)==0 )
+		if(dag_local_jobs_running(d)==0 && dag_remote_jobs_running(d)==0 && did_find_archived_job == 0 )
```

Comment:
A bit clunky, but I think it's the right solution.
Please add a comment here to explain why it is necessary.